Ds 2667

chschan opened the pull request.

chschan committed:
DS-2667: First draft of SegmentComparisonTable - spans  and fill color implemented

chschan committed:
Fix horizontal text alignment

chschan committed:
Compute p-values for numeric variables

chschan committed:
Use ComputeStats to calculate SEs, t-statistic and the n p-values

chschan committed:
Compute P-values for PickOne and PickAny using CellStatistic()

chschan committed:
Apply filters and weights consistently

chschan committed:
Fix problem caused by adding filters

chschan committed:
Use weights

chschan committed:
Make sure weights is always defined if required

chschan committed:
Compute index values

chschan committed:
Fix output formatting and documentation

chschan committed:
Fix documentation

chschan committed:
For categorical variables compute p-values using chi-square test instead of t-test if unweighted

chschan committed:
Fix p-values for numeric variables

chschan committed:
Simplify code from CellStatistics to only return P-values

chschan committed:
Fix documentation

chschan committed:
Fix row height

chschan committed:
Export function

chschan committed:
Fix missing values and variable set handling

chschan committed:
Fix handling of missing values

chschan committed:
Tidy up code and handle date variables

chschan committed:
Fix handling of dates and missing values

chschan committed:
Add tests

chschan committed:
Some more tidying up

chschan committed:
Rewrite tests to avoid errors in TravisCI

chschan committed:
Use sticky rows and use lower case for internal functions

chschan committed:
Adjust spacing

Commit:
DS-2667 Unit tests + ranking

Commit:
ds-2667 Rank tests

chschan committed:
Some formatting changes

chschan committed:
Fix calculations for cell fill

chschan committed:
Add option to remove NET/SUM rows

chschan committed:
Don't repeat variable name if it is the same as the question name

chschan committed:
Center numeric variables to determine fill color

chschan committed:
Update default format for output table

chschan committed:
Give error messages for unexpected inputs

jrwishart commented on R/segmentcomparisontable.R:
```diff
@@ -0,0 +1,289 @@
+        results.font.color <- matrix(font.color, nrow(result), ncol(result))
+        results.font.color[3:nrow(result),] <- font.color.nonsignificant # include p = NA
+        pvals <- matrix(NA, nrow(result), ncol(result))
+        rownames(pvals) <- row.labels
```
Code fails here due to a dimension mismatch when PickOneMulti input is used. See Routput throwing error in https://app.displayr.com/Dashboard?project_id=-705348#page=5a67bf81-8b62-4800-8cd1-64675dfba945

jrwishart commented on R/segmentcomparisontable.R:
```diff
@@ -0,0 +1,289 @@
+        row.span[[length(row.span) + 1]] <- list(label = attr(x[[vvi]], "question"), height = tmp.nvar)
+        if (!is.null(rownames(tmp)))
+            row.labels <- c(row.labels, rownames(tmp))
+        tmp.numeric <- isTRUE(attr(x[[vvi]], "questiontype") %in% c("NumberMulti", "Number"))
```
This might need "NumberGrid" as well? Seems to fail to capture the numeric variable type in https://app.displayr.com/Dashboard?project_id=-705348#page=5a67bf81-8b62-4800-8cd1-64675dfba945

jrwishart reviewed (changes requested):
Currently errors with some input variable types and numeric data isn't displaying as numeric (displaying as percentage). More information and reproducible examples in the comments.

chschan committed:
Fix handling of PickOneMulti and NumberGrid

jrwishart commented:
Changes have fixed the `PickOneMulti` and `NumberGrid` input type errors. However, the data reduction is still visible when the user toggles `Show NET or SUM`
![image](https://user-images.githubusercontent.com/4605715/69027451-11622800-0a23-11ea-8edc-4042fee7d8ae.png)
Example in https://app.displayr.com/Dashboard?project_id=-705348#page=5a67bf81-8b62-4800-8cd1-64675dfba945 and the variable type is `NumberGrid`

chschan committed:
Replace remove.net.or.sum with row.names.to.remove; Add more tests

chschan commented:
Fixed now. I have replaced the `Show NET or SUM` checkbox with a textbox `Rows to hide`, for consistency with the other Visualization/Table Standard R pages. The new form is on the page titled "Version 3": https://app.displayr.com/Dashboard?project_id=-705348#page=04606268-e538-422f-9e93-4d6ce626aca2
